sqlglot/dialects/tsql.py

```diff
@@ -3,9 +3,42 @@
 from sqlglot.generator import Generator
 from sqlglot.parser import Parser
 from sqlglot.tokens import Tokenizer, TokenType
+from sqlglot.helper import list_get
+from sqlglot.time import format_time
 
 
 class TSQL(Dialect):
+    date_name_mapping = {"weekday": "%A", "dw": "%A", "w": "%A", "month": "%b", "mm": "%b", "m": "%b"}
+    time_mapping = {
+        "yyyy": "%Y",
+        "yy": "%Y",
```

Comment:
is this correct?

Reply:
TSQL has different conventions in its functions:
DateName:
![image](https://user-images.githubusercontent.com/18576405/197143008-1dc7462f-96e7-4d75-8837-8f7de3da08b9.png)
DatePart:
![image](https://user-images.githubusercontent.com/18576405/197143132-c1886780-4713-42fe-9d86-bd1450653e29.png)

So I figured to create a default time_mapping which is then overwritten depending on which date function is used.
Please let me know if this is not desirable.